.travis.yml

```diff
@@ -36,11 +29,24 @@ cache:
   - "$HOME/.gradle/wrapper/"
 before_install:
 - ./travis/prepare-signing.sh $encrypted_677f232983c0_key $encrypted_677f232983c0_iv
-- if [ "$CROSSDOCK" == true ]; then bash ./travis/install-crossdock-deps.sh ; fi
-script:
-- if [ "$TESTS" == true ]; then make test-travis ; else echo 'skipping tests'; fi
-- if [ "$COVERAGE" == true ]; then ./gradlew codeCoverageReport ; else echo 'skipping coverage'; fi
-- if [ "$CROSSDOCK" == true ]; then bash ./travis/build-crossdock.sh ; else echo 'skipping crossdock'; fi
-after_success:
-- if [ "$COVERAGE" == true ]; then bash <(curl -s https://codecov.io/bash) ; fi
-- if [ "$CROSSDOCK" == true -a "$TRAVIS_PULL_REQUEST" == false ]; then ./gradlew upload ; fi
+
+stages:
+  - test
+  - name: deploy
+    if: env(TRAVIS_PULL_REQUEST) = false
+
+jobs:
+  include:
+    - stage: test
+      env: [ NAME=unit-test ]
+      script: make test-travis && ./gradlew codeCoverageReport && bash <(curl -s https://codecov.io/bash)
+
+    - env: [ NAME=crossdock ]
+      script: make crossdock-fresh
```

Comment:
Is the indentation correct here? I was expecting something like
```yaml
  - stage: test
    - env:
      script: 
    - env:
      script: 
```

Reply:
yes it is correct

Reply:
why is it correct though? It seems order dependent. Can we add `stage: test` as the first line of the second group, to make it more obvious?

Reply:
can try, I have copied it from this repo https://github.com/opentracing-contrib/java-spring-cloud/blob/master/.travis.yml#L25 it uses even more stages

Reply:
it looks like Travis explicitly allows that (i.e. the order matters and stage name is inherited from the previous item). Feels like a bad practice to me.

Reply:
This works (done in the last commit): 
```yml
jobs:
  include:
    - stage: test
      env: [ NAME=unit-test ]
      script: make test-travis && ./gradlew codeCoverageReport && bash <(curl -s https://codecov.io/bash)

    - stage: test
      env: [ NAME=crossdock ]
      script: make crossdock-fresh

    - stage: deploy
      env: [ NAME=deploy-to-maven ]
      script: ./gradlew upload

    - stage: deploy
      env: [ NAME=deploy-to-dockerhub ]
      script: ./travis/publish-crossdock.sh
```